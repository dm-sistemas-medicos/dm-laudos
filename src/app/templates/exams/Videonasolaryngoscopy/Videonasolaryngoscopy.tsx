"use client";

import NasalCavity from "../../components/NasalCavity/NasalCavity";
import Rhinopharynx from "../../components/Rhinopharynx/Rhinopharynx";
import Oropharynx from "../../components/Oropharynx/Oropharynx";
import Larynx from "../../components/Larynx/Larynx";
import Conclusion from "../../components/Conclusion/Conclusion";

import styles from "../exams.module.scss";

const Videonasolaryngoscopy = () => (
  <>
    <h1 className={styles["exam-title"]}>Videonasolaringoscopia</h1>
    <form>
      <NasalCavity />
      <Rhinopharynx />
      <Oropharynx />
      <Larynx />
      <Conclusion />
    </form>
  </>
);

export default Videonasolaryngoscopy;
