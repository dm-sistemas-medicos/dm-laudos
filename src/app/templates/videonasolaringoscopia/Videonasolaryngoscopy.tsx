"use client";

import { createContext, Dispatch, SetStateAction, useState } from "react";

import DragAndDrop from "../components/DragAndDrop/DragAndDrop";
import Preview from "../components/Preview/Preview";
import NasalCavity from "../components/NasalCavity/NasalCavity";
import Rhinopharynx from "../components/Rhinopharynx/Rhinopharynx";
import Oropharynx from "../components/Oropharynx/Oropharynx";
import Larynx from "../components/Larynx/Larynx";
import Conclusion from "../components/Conclusion/Conclusion";

import styles from "../layout.module.scss";

const MAX_ASIDE_IMAGES = 5;

type ImagesType = string[];
type SetImagesType = Dispatch<SetStateAction<ImagesType>>;

interface IImagesContext {
  images: ImagesType;
  setImages: SetImagesType;
}

const ImagesContext = createContext<IImagesContext>({
  images: [],
  setImages: () => {},
});

const Videonasolaryngoscopy = () => {
  const [images, setImages] = useState<ImagesType>([]);

  return (
    <ImagesContext.Provider value={{ images, setImages }}>
      <main className={styles.main}>
        <h1 className={styles["exam-title"]}>Videonasolaringoscopia</h1>
        <form>
          <NasalCavity />
          <Rhinopharynx />
          <Oropharynx />
          <Larynx />
          <Conclusion />
        </form>
      </main>
      <aside className={styles.aside}>
        <DragAndDrop setImages={setImages} />
        <Preview
          images={
            images.length <= MAX_ASIDE_IMAGES
              ? images
              : images.slice(0, MAX_ASIDE_IMAGES)
          }
          width={256}
          multiColumn={false}
        />
      </aside>
      {images.length > MAX_ASIDE_IMAGES && (
        <section className={styles.section}>
          <Preview
            images={images.slice(MAX_ASIDE_IMAGES)}
            width={256}
            multiColumn={true}
          />
        </section>
      )}
    </ImagesContext.Provider>
  );
};

export default Videonasolaryngoscopy;
