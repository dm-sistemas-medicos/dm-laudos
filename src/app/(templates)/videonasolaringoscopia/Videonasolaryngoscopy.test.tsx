import "@testing-library/jest-dom";

import { render } from "@testing-library/react";

import Videonasolaryngoscopy from "./Videonasolaryngoscopy";

describe("Videonasolaryngoscopy page (Client-side)", () => {
  it("renders the page", () => {
    const { container } = render(<Videonasolaryngoscopy />);

    expect(container).toMatchSnapshot();
  });
});
