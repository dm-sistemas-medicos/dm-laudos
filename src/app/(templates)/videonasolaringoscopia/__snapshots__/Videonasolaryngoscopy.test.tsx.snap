// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Videonasolaryngoscopy page (Client-side) renders the page 1`] = `
<div>
  <h1
    class="exam-title"
  >
    Videonasolaringoscopia
  </h1>
  <form>
    <fieldset>
      <h2
        class="exam-section-title"
      >
        Cavidade Nasal
      </h2>
      <ul
        class="exam-item-list"
      >
        <li
          class="exam-item"
          data-testid="nasal-cavity-nasal-septum-container"
        >
          <label
            for="nasal-cavity-nasal-septum"
          >
            Septo nasal
          </label>
          <textarea
            data-testid="nasal-cavity-nasal-septum"
            id="nasal-cavity-nasal-septum"
            rows="2"
          >
            com desvio para direita, com esporão ósseo associado, parcialmente obstrutivo
          </textarea>
        </li>
        <li
          class="exam-item"
          data-testid="nasal-cavity-middle-turbinates-container"
        >
          <label
            for="nasal-cavity-middle-turbinates"
          >
            Cornetos médios
          </label>
          <textarea
            data-testid="nasal-cavity-middle-turbinates"
            id="nasal-cavity-middle-turbinates"
            rows="1"
          >
            normotróficos
          </textarea>
        </li>
        <li
          class="exam-item"
          data-testid="nasal-cavity-inferior-turbinates-container"
        >
          <label
            for="nasal-cavity-inferior-turbinates"
          >
            Cornetos inferiores
          </label>
          <textarea
            data-testid="nasal-cavity-inferior-turbinates"
            id="nasal-cavity-inferior-turbinates"
            rows="1"
          >
            hipertróficos
          </textarea>
        </li>
        <li
          class="exam-item"
          data-testid="nasal-cavity-secretion-container"
        >
          <label
            for="nasal-cavity-secretion"
          >
            Secreção
          </label>
          <textarea
            data-testid="nasal-cavity-secretion"
            id="nasal-cavity-secretion"
            rows="1"
          >
            aquosa
          </textarea>
        </li>
        <li
          class="exam-item"
          data-testid="nasal-cavity-mucosa-container"
        >
          <label
            for="nasal-cavity-mucosa"
          >
            Mucosa
          </label>
          <textarea
            data-testid="nasal-cavity-mucosa"
            id="nasal-cavity-mucosa"
            rows="1"
          >
            pálida
          </textarea>
        </li>
        <li
          class="exam-item non-printable"
          data-testid="nasal-cavity-comments-container"
        >
          <textarea
            class="exam-item-double"
            data-testid="nasal-cavity-comments"
            id="nasal-cavity-comments"
            placeholder="Observações"
            rows="1"
          />
        </li>
      </ul>
    </fieldset>
    <fieldset>
      <h2
        class="exam-section-title"
      >
        Rinofaringe
      </h2>
      <ul
        class="exam-item-list"
      >
        <li
          class="exam-item"
          data-testid="rhinopharynx-mucosa-container"
        >
          <label
            for="rhinopharynx-mucosa"
          >
            Mucosa
          </label>
          <textarea
            data-testid="rhinopharynx-mucosa"
            id="rhinopharynx-mucosa"
            rows="1"
          >
            normal
          </textarea>
        </li>
        <li
          class="exam-item"
          data-testid="rhinopharynx-meatus-container"
        >
          <label
            for="rhinopharynx-meatus"
          >
            Meatos
          </label>
          <textarea
            data-testid="rhinopharynx-meatus"
            id="rhinopharynx-meatus"
            rows="1"
          >
            pérvios
          </textarea>
        </li>
        <li
          class="exam-item"
          data-testid="rhinopharynx-tubal-ostia-container"
        >
          <label
            for="rhinopharynx-tubal-ostia"
          >
            Óstios tubários
          </label>
          <textarea
            data-testid="rhinopharynx-tubal-ostia"
            id="rhinopharynx-tubal-ostia"
            rows="1"
          >
            visualidados bilateralmente e pérvios
          </textarea>
        </li>
        <li
          class="exam-item"
          data-testid="rhinopharynx-velopharyngeal-function-container"
        >
          <label
            for="rhinopharynx-velopharyngeal-function"
          >
            Função velofaringea
          </label>
          <textarea
            data-testid="rhinopharynx-velopharyngeal-function"
            id="rhinopharynx-velopharyngeal-function"
            rows="1"
          >
            preservada
          </textarea>
        </li>
        <li
          class="exam-item"
          data-testid="rhinopharynx-injuries-container"
        >
          <textarea
            aria-label="Lesões"
            class="exam-item-double"
            data-testid="rhinopharynx-injuries"
            id="rhinopharynx-injuries"
            rows="1"
          >
            Ausência de lesões
          </textarea>
        </li>
        <li
          class="exam-item non-printable"
          data-testid="rhinopharynx-comments-container"
        >
          <textarea
            class="exam-item-double"
            data-testid="rhinopharynx-comments"
            id="rhinopharynx-comments"
            placeholder="Observações"
            rows="1"
          />
        </li>
      </ul>
    </fieldset>
    <fieldset>
      <h2
        class="exam-section-title"
      >
        Orofaringe
      </h2>
      <ul
        class="exam-item-list"
      >
        <li
          class="exam-item"
          data-testid="oropharynx-oropharynx-container"
        >
          <label
            for="oropharynx-oropharynx"
          >
            Orofaringe
          </label>
          <textarea
            data-testid="oropharynx-oropharynx"
            id="oropharynx-oropharynx"
            rows="1"
          >
            e hipofaringe com mucosa normal
          </textarea>
        </li>
        <li
          class="exam-item"
          data-testid="oropharynx-base-of-tongue-container"
        >
          <label
            for="oropharynx-base-of-tongue"
          >
            Base da língua
          </label>
          <textarea
            data-testid="oropharynx-base-of-tongue"
            id="oropharynx-base-of-tongue"
            rows="1"
          >
            sem alterações
          </textarea>
        </li>
        <li
          class="exam-item"
          data-testid="oropharynx-vallecula-pyriform-sinuses-container"
        >
          <label
            for="oropharynx-vallecula-pyriform-sinuses"
          >
            Valéculas e seios piriformes
          </label>
          <textarea
            data-testid="oropharynx-vallecula-pyriform-sinuses"
            id="oropharynx-vallecula-pyriform-sinuses"
            rows="1"
          >
            livres
          </textarea>
        </li>
        <li
          class="exam-item non-printable"
          data-testid="oropharynx-comments-container"
        >
          <textarea
            class="exam-item-double"
            data-testid="oropharynx-comments"
            id="oropharynx-comments"
            placeholder="Observações"
            rows="1"
          />
        </li>
      </ul>
    </fieldset>
    <fieldset>
      <h2
        class="exam-section-title"
      >
        Laringe
      </h2>
      <ul
        class="exam-item-list"
      >
        <li
          class="exam-item"
          data-testid="larynx-vocal-folds-container"
        >
          <label
            for="larynx-vocal-folds"
          >
            Pregas vocais
          </label>
          <textarea
            data-testid="larynx-vocal-folds"
            id="larynx-vocal-folds"
            rows="1"
          >
            com mobilidade preservada bilateralmente
          </textarea>
        </li>
        <li
          class="exam-item"
          data-testid="larynx-free-edges-container"
        >
          <label
            for="larynx-free-edges"
          >
            Bordos livres
          </label>
          <textarea
            data-testid="larynx-free-edges"
            id="larynx-free-edges"
            rows="1"
          >
            regulares, sem lesões
          </textarea>
        </li>
        <li
          class="exam-item"
          data-testid="larynx-coaptation-container"
        >
          <label
            for="larynx-coaptation"
          >
            Coaptação
          </label>
          <textarea
            data-testid="larynx-coaptation"
            id="larynx-coaptation"
            rows="1"
          >
            normal
          </textarea>
        </li>
        <li
          class="exam-item"
          data-testid="larynx-mucosa-container"
        >
          <label
            for="larynx-mucosa"
          >
            Mucosa
          </label>
          <textarea
            data-testid="larynx-mucosa"
            id="larynx-mucosa"
            rows="1"
          >
            com hiperemia e edema leve
          </textarea>
        </li>
        <li
          class="exam-item"
          data-testid="larynx-subglottis-container"
        >
          <label
            for="larynx-subglottis"
          >
            Subglote
          </label>
          <textarea
            data-testid="larynx-subglottis"
            id="larynx-subglottis"
            rows="1"
          >
            livre, sem lesões
          </textarea>
        </li>
        <li
          class="exam-item non-printable"
          data-testid="larynx-comments-container"
        >
          <textarea
            class="exam-item-double"
            data-testid="larynx-comments"
            id="larynx-comments"
            placeholder="Observações"
            rows="1"
          />
        </li>
      </ul>
    </fieldset>
    <fieldset
      data-testid="conclusion-container"
    >
      <h2
        class="exam-section-title"
      >
        Conclusão
      </h2>
      <ul
        class="exam-item-list"
      >
        <li
          class="exam-item"
        >
          <textarea
            aria-label="Conclusão"
            class="exam-item-double"
            data-testid="conclusion"
            id="conclusion"
            rows="5"
          />
        </li>
      </ul>
    </fieldset>
  </form>
</div>
`;
